python/pyspark/sql/tests/test_types.py

```diff
@@ -753,6 +753,9 @@ def test_cast_to_string_with_udt(self):
         result = df.select(F.col("point").cast("string"), F.col("pypoint").cast("string")).head()
         self.assertEqual(result, Row(point="(1.0, 2.0)", pypoint="[3.0, 4.0]"))
 
+    @unittest.skipIf(
+        "SPARK_SKIP_CONNECT_COMPAT_TESTS" in os.environ, "SPARK-49787: Supported since Spark 4.0.0"
```

Comment:
I'm a bit confused, I think CAST between UDTs is a long-standing feature, and this test was added quite a while ago. Why it suddenly failed?

Reply:
To @cloud-fan , the failure happens after the following as mentioned in the PR.
- #48251

AFAIK, it was not supported in `branch-3.4`. So, it fails during the compat test.

Although you reverted it, you will introduce the following right? Are you going to backport it to branch-3.4? If then, we can revert this PR after merging your PR to branch-3.4.
- #48881

Reply:
Yes. And, it happens because the different functionality between Spark 4.0 and 3.5.
> so PRs merged into the master branch may break tests in branch 3.5?

There are the existing examples of behavior differences.
```
python/pyspark/sql/tests/test_dataframe.py:        "SPARK_SKIP_CONNECT_COMPAT_TESTS" in os.environ, "Newline difference from the server"
python/pyspark/sql/tests/test_readwriter.py:        "SPARK_SKIP_CONNECT_COMPAT_TESTS" in os.environ, "Known behavior change in 4.0"
python/pyspark/sql/tests/test_types.py:        "SPARK_SKIP_CONNECT_COMPAT_TESTS" in os.environ, "SPARK-49787: Supported since Spark 4.0.0"
```